### PR TITLE
New VS121 Version.

### PR DESCRIPTION
**Pull request by J2thatsme**
Added new functionalities for the new VS121 Version. Update is backwards compatible.

### CODE REVIEW

File: types/milesight/VS121/uplink.js [hunk 1]
```diff
@@ -91,26 +97,114 @@ function consume(event) {
...
+      decoded.region = bytes[i];
+      decoded.dwellTimeAvg = readUInt16LE(bytes.slice(i + 1, i + 3));
+      decoded.dwellTimeMax = readUInt16LE(bytes.slice(i + 3, i + 5));
+      i += 5;
+      emit("sample", { data: decoded, topic: "dwell_time" });
+    }
+    // TIMESTAMP
+    else if (channelId === 0x0f && channelType === 0x85) {
+      /*
+      // We probably do not need that atm
```
Comment (mariusgiger) [inline types/milesight/VS121/uplink.js:193]: use this as the sample timestamp?

File: types/milesight/VS121/a_flow.schema.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{
+  "$id": "https://akenza.io/xovis/VS121/a_flow.schema.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "processingType": "uplink_decoder",
+  "topic": "a_flow",
+  "title": "A flow",
```
Comment (mariusgiger) [inline types/milesight/VS121/a_flow.schema.json:7]: what does this mean?

File: types/milesight/VS121/dwell_time.schema.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  "$id": "https://akenza.io/xovis/VS121/dwell_time.schema.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "processingType": "uplink_decoder",
+  "topic": "dwell_time",
+  "title": "Dwell time",
+  "properties": {
+    "region": {
```
Comment (mariusgiger) [inline types/milesight/VS121/dwell_time.schema.json:9]: possibly add a topic per region to make aggregations easier

File: types/milesight/VS121/region_counter.schema.json [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      "description": "Region 1 count",
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 255
+    },
+    "region2count": {
+      "title": "Region 2 count",
+      "description": "Region 2 count",
+      "type": "integer",
+      "minimum": 0,
```
Comment (mariusgiger) [inline types/milesight/VS121/region_counter.schema.json:20]: try to add units if possible

File: types/milesight/VS121/region_counter.schema.json [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+      "maximum": 255
+    },
+    "region3count": {
+      "title": "Region 3 count",
+      "description": "Region 3 count",
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 255
+    },
+    "region4count": {
```
Comment (mariusgiger) [inline types/milesight/VS121/region_counter.schema.json:30]: possibly rename key to match region4Count to align with camelCase

**Review by mariusgiger — CHANGES_REQUESTED** [review]
Thanks! I have added a few inputs regarding data structuring

**Comment by J2thatsme** [comment]
Changes are included in the VS133 Branch

**Pull request closed without merging**